apps/prairielearn/python/question_phases.py

```diff
@@ -165,7 +165,7 @@ def process_element(
                     # TODO: Once this has been running in production for a while,
                     # change this to raise an exception.
                     sys.stderr.write(
-                        f"Function {str(phase)}() in {str(element_controller)} returned a data object other than the one that was passed in.\n\n"
+                        f"Function {phase!s}() in {element_controller!s} returned a data object other than the one that was passed in.\n\n"
```

Comment:
This format specifier forces a string conversion.

Reply:
From https://docs.astral.sh/ruff/rules/explicit-f-string-type-conversion/:

> Note that, in many cases, calling str() within an f-string is unnecessary and can be removed entirely, as the value will be converted to a string automatically, the notable exception being for classes that implement a custom __format__ method.

What's your thinking on this? e.g. in this situation, `phase` and `element_controller` are both already strings, so do we gain anything with an explicit conversion? Ditto elsewhere in the file, I think we should consider on a case-by-case basis if `!s` is needed or if we can rely on default string conversion.

Reply:
This has since been resolved; we're avoiding the use of `!s` when possible.